src/nvim/eval.c

```diff
@@ -21757,3 +21971,81 @@ bool eval_has_provider(char *name)
 
   return false;
 }
+
+static DictWatcher *dictwatcher_node_data(QUEUE *q)
+{
+  return QUEUE_DATA(q, DictWatcher, node);
+}
+
+static void dictwatcher_notify(dict_T *dict, const char *key,
+    typval_T *newtv, typval_T *oldtv)
+{
+  typval_T argv[2];
+  for (size_t i = 0; i < ARRAY_SIZE(argv); i++) {
+    init_tv(argv + i);
+  }
+
+  argv[0].v_type = VAR_STRING;
+  argv[0].vval.v_string = (uint8_t *)xstrdup(key);
```

Comment:
I thought we would be using C99 builtin types whenever possible? I've been replacing `char_u` by `uint8_t` every time I can.